management/vmutils/rolestate.go

```diff
@@ -0,0 +1,31 @@
+package vmutils
+
+import (
+	"time"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management"
+	vm "github.com/MSOpenTech/azure-sdk-for-go/management/virtualmachine"
+)
+
+// Blocks until all role instances in deployment to reach desired power state
+func WaitForDeploymentPowerState(client management.Client, cloudServiceName, deploymentName string, desiredPowerstate vm.PowerState) error {

```

Comment:
@paulmey I see. Yeah PowerState is not the thing users go for usually. It'd be great to actually watch for RoleState.